### PR TITLE
Set up CI

### PR DESCRIPTION
**Pull request by robintown**
Let's start using GitHub Actions to check formatting/lints and ensure that all generated files are up to date.

**Based on https://github.com/element-hq/compound-design-tokens/pull/87** - please only review the later commits

### CODE REVIEW

File: assets/web/js/cpdLightHc.js [hunk 1]
```diff
@@ -56,18 +56,18 @@ export const cpdFontLetterSpacingHeadingSm = "-0.0166em";
...
+export const cpdFontBodyXsSemibold = "600 0.688rem/1.5 Inter";
+export const cpdFontBodySmRegular = "400 0.813rem/1.5 Inter";
+export const cpdFontBodySmMedium = "500 0.813rem/1.5 Inter";
+export const cpdFontBodySmSemibold = "600 0.813rem/1.5 Inter";
+export const cpdFontBodyMdRegular = "400 0.938rem/1.5 Inter";
+export const cpdFontBodyMdMedium = "500 0.938rem/1.5 Inter";
+export const cpdFontBodyMdSemibold = "600 0.938rem/1.5 Inter";
+export const cpdFontBodyLgRegular = "400 1.063rem/1.5 Inter";
+export const cpdFontBodyLgMedium = "500 1.063rem/1.5 Inter";
+export const cpdFontBodyLgSemibold = "600 1.063rem/1.5 Inter";
```
Comment (jmartinesp) [inline assets/web/js/cpdLightHc.js:70]: These actually change the size, even if not by much. Is that ok?

**Review by jmartinesp — APPROVED** [review]
LGTM, just a comment about a potential minor issue.

File: assets/web/js/cpdLightHc.js (hunk 1 above)
Comment (jmartinesp) [reply assets/web/js/cpdLightHc.js:70]: Oh, this should actually go into https://github.com/element-hq/compound-design-tokens/pull/87 😅

**Comment by jmartinesp** [comment]
I changed the base to  the `lint` branch, it should make reviewing easier and GH will automatically move it to `main` once that branch is merged.

File: .github/workflows/automation.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Build generated files
+        run: yarn build
+      - name: Check for changes
+        run: >
+          if [ -z "$(git status --porcelain)" ];
+          then
+            echo "Generated files are up to date."
+            exit 0
+          else
+            echo "Some files have not been regenerated recently. Please run 'yarn build' to fix this."
```
Comment (bmarty) [inline .github/workflows/automation.yaml:58]:
```suggestion
            echo "The files below have not been regenerated. Please run 'yarn build' to fix this."
```

File: .github/workflows/automation.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b # v4
+        with:
+          cache: yarn
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Lint
+        run: yarn exec biome -- ci
+
+  build:
+    name: Build generated files
```
Comment (bmarty) [inline .github/workflows/automation.yaml:38]:
```suggestion
    name: Check that generated files are up to date
```

File: .github/workflows/automation.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          cache: yarn
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Build generated files
+        run: yarn build
+      - name: Check for changes
+        run: >
+          if [ -z "$(git status --porcelain)" ];
+          then
+            echo "Generated files are up to date."
```
Comment (bmarty) [inline .github/workflows/automation.yaml:55]: Thanks for adding this check, I was about to suggest to do it!

**Review by bmarty — APPROVED** [review]
Thanks!

File: assets/web/js/cpdLightHc.js (hunk 1 above)
Comment (robintown) [reply assets/web/js/cpdLightHc.js:70]: Ah, I missed this before I merged the other PR. I'll try to research why Style Dictionary is rounding the values, and whether it messes with anything.

Comment (robintown) [reply assets/web/js/cpdLightHc.js:70]: (as in, I'll do that in another PR now that it's already too late…)

**Pull request merged**